hr_course/models/hr_course.py

```diff
@@ -180,6 +183,41 @@ def cancel_course(self):
         for record in self:
             record.write(record._cancel_course_values())
 
+    @api.model
+    def send_course_notification_email(self, partner_emails):
+        email_list = partner_emails.mapped("email")
+        company = self.env["res.company"].browse(
+            self.env.context.get("company_id") or self.env.company.id
+        )
+        if email_list:
+            email_template = self.env.ref(
+                "hr_course.mail_template_validity_reminder_en"
+            )
+            for email in email_list:
+                options = {
+                    "email_to": email.strip(),
+                    "email_from": company.email.strip(),
+                }
+                email_template.send_mail(self.id, email_values=options, force_send=True)
+                return True
+
+    @api.model
+    def process_validity(self):
+        company_id = self.env.context.get("company_id") or self.env.company.id
+        alerting_delay = self.env["res.company"].browse(company_id).alerting_delay
+        mailing_list = self.env["res.company"].browse(company_id).mailing_list_to_alert
+        for course in self:
+            if course.validity_end_date:
+                if course.validity_end_date >= (
+                    fields.Date.today() - timedelta(days=alerting_delay)
```

Comment:
are you sure about the computation here? 

If today is 2021-12-22, I expect to be warned for courses which expire before 2021-12-22 + 2days, so the test should be `course.validity_end_date - alerting_delay <= today`